test/unit/event_store/event_test.rb

```diff
@@ -87,38 +87,47 @@ def test_stale
     assert_equal expected_stale_events_count, EventStore::Event.stale.count
   end
 
-  def test_rolledback_harmless_exceptions_not_reported
-    event = EventStore::Event.create!(stream: 'dummy', event_type: 'Dummy', event_id: 1, provider_id: 1, data: { whatever: 1 })
+  class EventRollbacksTest < ActiveSupport::TestCase
+    disable_transactional_fixtures!
```

Comment:
why was this needed?

Reply:
So, the 2nd test actually was failing, because the ErrorReporting was never called.
This was because no error was caught when calling `rolledback!`, and this, in its turn was because the `deserialize` method (which throws the NotFound) is only called when Rails tries to restore the state of the object, which was actually not happening. I guess the reason is that it was not a real transaction so [this condition](https://github.com/rails/rails/blob/v6.1.7.7/activerecord/lib/active_record/transactions.rb#L402) was never fulfilled.

When I wrapped this within a real transaction, it still wasn't working, I guess because `restore_state[:level]` [here](https://github.com/rails/rails/blob/v6.1.7.7/activerecord/lib/active_record/transactions.rb#L403) was equal to 2. I didn't look deeper what it means, but I just thought it could be because each test by default creates a transaction on its own, so this new transaction within test is "level 2". So, I tried disabling transactional tests, and -boom- the test passed :sweat_smile: 

I don't know whether it's the right solution, but it seems to be working :grimacing:

Reply:
strange, tests all worked locally in my PR, I'll check it out tomorrow